golang/thrift/outbound.go

```diff
@@ -0,0 +1,323 @@
+	return p.writer.WriteFieldEnd()
+}
+
+// WriteFieldStop delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteFieldStop() error {
+	return p.writer.WriteFieldStop()
+}
+
+// WriteMapBegin delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteMapBegin(keyType thrift.TType, valueType thrift.TType, size int) error {
+	return p.writer.WriteMapBegin(keyType, valueType, size)
+}
+
+// WriteMapEnd delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteMapEnd() error {
+	return p.writer.WriteMapEnd()
+}
+
+// WriteListBegin delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteListBegin(elemType thrift.TType, size int) error {
+	return p.writer.WriteListBegin(elemType, size)
+}
+
+// WriteListEnd delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteListEnd() error {
+	return p.writer.WriteListEnd()
+}
+
+// WriteSetBegin delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteSetBegin(elemType thrift.TType, size int) error {
+	return p.writer.WriteSetBegin(elemType, size)
+}
+
+// WriteSetEnd delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteSetEnd() error {
+	return p.writer.WriteSetEnd()
+}
+
+// WriteBool delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteBool(value bool) error {
+	return p.writer.WriteBool(value)
+}
+
+// WriteByte delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteByte(value byte) error {
+	return p.writer.WriteByte(value)
+}
+
+// WriteI16 delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteI16(value int16) error {
+	return p.writer.WriteI16(value)
+}
+
+// WriteI32 delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteI32(value int32) error {
+	return p.writer.WriteI32(value)
+}
+
+// WriteI64 delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteI64(value int64) error {
+	return p.writer.WriteI64(value)
+}
+
+// WriteDouble delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteDouble(value float64) error {
+	return p.writer.WriteDouble(value)
+}
+
+// WriteString delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteString(value string) error {
+	return p.writer.WriteString(value)
+}
+
+// WriteBinary delegates to the TBinaryProtocol writer
+func (p *TChannelOutboundProtocol) WriteBinary(value []byte) error {
+	return p.writer.WriteBinary(value)
+}
+
+// Flush takes the written content from the write buffer and writes
+// it as arg3 to the underlying tchannel.
+func (p *TChannelOutboundProtocol) Flush() error {
+	// flush to write buffer
+	if err := p.writer.Flush(); err != nil {
+		return err
+	}
+	payload := p.writeBuffer.Bytes()
+
+	// begin the outbound call
+	ctx, _ := context.WithTimeout(context.Background(), time.Second*time.Duration(1))
```

Comment:
The timeout needs to be controllable by the calling code.  In fact the entire context needs to be passed from the caller, or else you won't be able to do zipkin trace flags propagation, request deadline propagation, etc.  I'd recommend passing the context into NewTChannelOutboundProtocol.